inst/include/child_model_simulation.hpp

```diff
@@ -76,65 +78,78 @@ void run_wlhiv_births(int t,
+                   (n_ba.h_art_adult(ht, hd, a, 1) + c_ba.h_art_adult(ht, hd, a, 1)) / 2;
+      } // end hTS
     } // end hDS
 
 
-    if (intermediate.children.nHIVcurr > 0) {
-      intermediate.children.df = intermediate.children.df / ((intermediate.children.nHIVcurr + intermediate.children.nHIVlast) / 2);
+    if (i_hc.nHIVcurr > 0) {
+      auto mean_nHIV = (i_hc.nHIVcurr + i_hc.nHIVlast) / 2;
+      i_hc.df = i_hc.df / mean_nHIV;
     } else {
-      intermediate.children.df = 1;
+      i_hc.df = 1;
     }
 
-    for (int hd = 0; hd < ss.hDS; ++hd) {
-      intermediate.children.df += cpars.local_adj_factor * cpars.fert_mult_by_age(a) * cpars.fert_mult_off_art(hd) * ((state_next.base.h_hiv_adult(hd, a, 1) + state_curr.base.h_hiv_adult(hd, a, 1)) / 2);
-      //women on ART less than 6 months use the off art fertility multiplier
-      intermediate.children.df += cpars.local_adj_factor * cpars.fert_mult_by_age(a) * cpars.fert_mult_off_art(hd) * ((state_next.base.h_art_adult(0, hd, a, 1) + state_curr.base.h_art_adult(0, hd, a, 1)) / 2);
-      for (int ht = 1; ht < ss.hTS; ++ht) {
-        intermediate.children.df += cpars.local_adj_factor * cpars.fert_mult_on_art(a) * ((state_next.base.h_art_adult(ht, hd, a, 1) + state_curr.base.h_art_adult(ht, hd, a, 1)) / 2);
-      } //end hTS
+    for (int hd = 0; hd < ss_b.hDS; ++hd) {
+      i_hc.df += p_hc.local_adj_factor * p_hc.fert_mult_by_age(a) * p_hc.fert_mult_off_art(hd) *
+                 (n_ba.h_hiv_adult(hd, a, 1) + c_ba.h_hiv_adult(hd, a, 1)) / 2;
+      // women on ART less than 6 months use the off art fertility multiplier
+      i_hc.df += p_hc.local_adj_factor * p_hc.fert_mult_by_age(a) * p_hc.fert_mult_off_art(hd) *
+                 (n_ba.h_art_adult(0, hd, a, 1) + c_ba.h_art_adult(0, hd, a, 1)) / 2;
+      for (int ht = 1; ht < ss_b.hTS; ++ht) {
+        i_hc.df += p_hc.local_adj_factor * p_hc.fert_mult_on_art(a) *
+                   (n_ba.h_art_adult(ht, hd, a, 1) + c_ba.h_art_adult(ht, hd, a, 1)) / 2;
+      } // end hTS
     } // end hDS
 
-    intermediate.children.birthsCurrAge = (intermediate.children.nHIVcurr + intermediate.children.nHIVlast) / 2 * cpars.total_fertility_rate(t) * intermediate.children.df / (intermediate.children.df * intermediate.children.prev + 1 - intermediate.children.prev) *  demog.age_specific_fertility_rate(a, t) / intermediate.children.asfr_sum ;
-    intermediate.children.birthsHE += intermediate.children.birthsCurrAge;
+    auto mean_nHIV = (i_hc.nHIVcurr + i_hc.nHIVlast) / 2;
+    i_hc.birthsCurrAge = mean_nHIV * p_hc.total_fertility_rate(t) *
+                         i_hc.df / (i_hc.df * i_hc.prev + 1 - i_hc.prev) *
+                         p_dm.age_specific_fertility_rate(a, t) / i_hc.asfr_sum ;
+    i_hc.birthsHE += i_hc.birthsCurrAge;
     if (a < 9) {
-      intermediate.children.births_HE_15_24 += intermediate.children.birthsCurrAge;
+      i_hc.births_HE_15_24 += i_hc.birthsCurrAge;
```

Comment:
Comment unrelated to this PR:

@mwalte10 -- switch to camel case here vs. snake case generally is slightly distracting. Is this an artefact of transcribing some things out of Spectrum?

I'm okay to leave it as is for now unless anyone else feels strongly.

Reply:
i dont mind doing it/you doing it but it will require changes to the intermediate state stuff so i dont want to do it in this pr

Reply:
as discussed ill pick this up